dispatcher.go

```diff
@@ -0,0 +1,164 @@
+package gofc
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/nutanix/gofc/ofprotocol/ofp13"
+)
+
+type Dispatcher struct {
+	server *OFController
+	mtx    sync.Mutex
+}
+
+func NewDispatcher(s *OFController) *Dispatcher {
+	d := new(Dispatcher)
+	d.server = s
+	return d
+}
+
+func (d *Dispatcher) handleMessage(msg ofp13.OFMessage, dp *Datapath) {
+	d.mtx.Lock()
+	defer func() { d.mtx.Unlock() }()
```

Comment:
Use `defer d.mtx.Unlock()`. It will bind `d.mtx` here (where it is clear that it matches `d.mtx.Lock`) instead of at function panic/return.

Reply:
Fixed